presto-native-execution/presto_cpp/main/types/FunctionMetadata.cpp

```diff
@@ -0,0 +1,222 @@
+getScalarFunctionMetadata(
+    const std::string& functionName,
+    const std::vector<const FunctionSignature*>& functionSignatures) {
+  std::vector<protocol::JsonBasedUdfFunctionMetadata>
+      jsonBasedUdfFunctionMetadataList;
+  jsonBasedUdfFunctionMetadataList.reserve(functionSignatures.size());
+  const protocol::FunctionKind functionKind = protocol::FunctionKind::SCALAR;
+
+  for (const auto& functionSignature : functionSignatures) {
+    protocol::JsonBasedUdfFunctionMetadata jsonBasedUdfFunctionMetadata;
+    jsonBasedUdfFunctionMetadata.functionKind = functionKind;
+    jsonBasedUdfFunctionMetadata.routineCharacteristics =
+        getRoutineCharacteristics(
+            *functionSignature, functionName, functionKind);
+
+    updateFunctionMetadata(
+        functionName, *functionSignature, jsonBasedUdfFunctionMetadata);
+    jsonBasedUdfFunctionMetadataList.emplace_back(jsonBasedUdfFunctionMetadata);
+  }
+  return jsonBasedUdfFunctionMetadataList;
+}
+
+const std::vector<protocol::JsonBasedUdfFunctionMetadata>
+getWindowFunctionMetadata(
+    const std::string& functionName,
+    const std::vector<FunctionSignaturePtr>& windowFunctionSignatures) {
+  std::vector<protocol::JsonBasedUdfFunctionMetadata>
+      jsonBasedUdfFunctionMetadataList;
+  jsonBasedUdfFunctionMetadataList.reserve(windowFunctionSignatures.size());
+  const protocol::FunctionKind functionKind = protocol::FunctionKind::WINDOW;
+
+  for (const auto& windowFunctionSignature : windowFunctionSignatures) {
+    protocol::JsonBasedUdfFunctionMetadata jsonBasedUdfFunctionMetadata;
+    jsonBasedUdfFunctionMetadata.functionKind = functionKind;
+    jsonBasedUdfFunctionMetadata.routineCharacteristics =
+        getRoutineCharacteristics(
+            *windowFunctionSignature, functionName, functionKind);
+
+    updateFunctionMetadata(
+        functionName, *windowFunctionSignature, jsonBasedUdfFunctionMetadata);
+    jsonBasedUdfFunctionMetadataList.emplace_back(jsonBasedUdfFunctionMetadata);
+  }
+  return jsonBasedUdfFunctionMetadataList;
+}
+
+const std::vector<protocol::JsonBasedUdfFunctionMetadata> getFunctionMetadata(
+    const std::string& functionName) {
+  if (auto aggregateFunctionSignatures =
+          getAggregateFunctionSignatures(functionName)) {
+    return getAggregateFunctionMetadata(
+        functionName, aggregateFunctionSignatures.value());
+  } else if (
+      auto windowFunctionSignatures =
+          getWindowFunctionSignatures(functionName)) {
+    return getWindowFunctionMetadata(
+        functionName, windowFunctionSignatures.value());
+  } else {
+    auto functionSignatures = getFunctionSignatures();
+    if (functionSignatures.find(functionName) != functionSignatures.end()) {
+      return getScalarFunctionMetadata(
+          functionName, functionSignatures.at(functionName));
+    }
+  }
+  VELOX_UNREACHABLE("Function kind cannot be determined");
+}
+
+} // namespace
+
+void getJsonMetadataForFunction(
```

Comment:
This seems like a good internal function to use in getJsonFunctionMetadata. But it might be better to expose a function that returns a json object instead of adding it to a list for the tests. The above getFunctionMetadata is sufficient.

Reply:
Updated the function to return a json instead of passing the list as an argument. This is not kept internal though as it is also used in the tests, would this be fine?